dev/fattest.simplicity/src/componenttest/topology/utils/MvnUtils.java

```diff
@@ -70,296 +69,365 @@
+                                   Set<String> versionedJars) throws Exception {
+        MvnUtils mvn = new MvnUtils(server, bucketName, testName, suiteFileName, additionalProps, versionedJars);
+        return mvn.runCleanTest();
+    }
 
-        String mvn = "mvn";
-        if (System.getProperty("os.name").contains("Windows")) {
-            mvn = mvn + ".cmd";
-        }
-        mvnCliRaw = new String[] { mvn, "clean", "test", "-Dwlp=" + wlp, "-Dtck_server=" + server.getServerName(),
-                                   "-Dtck_failSafeUndeployment=" + DEFAULT_FAILSAFE_UNDEPLOYMENT,
-                                   "-Dtck_appDeployTimeout=" + DEFAULT_APP_DEPLOY_TIMEOUT,
-                                   "-Dtck_appUndeployTimeout=" + DEFAULT_APP_UNDEPLOY_TIMEOUT,
-                                   "-Dtck_port=" + server.getPort(PortType.WC_defaulthost),
-                                   "-DtargetDirectory=" + getResultsDir() + "/" + getTargetFolderName(),
-                                   "-DcomponentRootDir=" + componentRootDir,
-                                   "-Dsun.rmi.transport.tcp.responseTimeout=" + DEFAULT_MBEAN_TIMEOUT
-        };
-
-        mvnCliRoot = concatStringArray(mvnCliRaw, getJarCliProperties(server, jarsFromWlp));
+    /**
+     * Full constructor for MvnUtils. In most cases one of the static convenience methods should be used instead of calling this directly.
+     *
+     * @param server the liberty server which should be used to run the TCK
+     * @param bucketName the name of the test project
+     * @param testName the name of the method that's being used to launch the TCK
+     * @param suiteFileName the name of the suite xml file
+     * @param additionalProps java properties to set when running the mvn command
+     * @param versionedJars A set of versioned jars
+     */
+    public MvnUtils(LibertyServer server, String bucketName, String testName, String suiteFileName, Map<String, String> additionalMvnProps,
+                    Set<String> versionedJars) {
+        this.server = server;
+        this.suiteFileName = suiteFileName;
+        this.bucketName = bucketName;
+        this.testName = testName;
+        this.versionedJars = versionedJars;
+        this.additionalMvnProps = additionalMvnProps;
+    }
 
-        // The cmd below is a base for running the TCK as a whole for this project.
-        // It is possible to use other Testng control xml files (and even generate them
-        // based on examining the TCK jar) in which case the value for suiteXmlFile would
-        // be different.
-        String suiteFile = overrideSuiteFileName == null ? defaultSuiteFile : overrideSuiteFileName;
-        mvnCliTckRoot = concatStringArray(mvnCliRoot, new String[] { "-DsuiteXmlFile=" + suiteFile });
+    /**
+     * Get a list of Files which represent JUnit xml results files.
+     *
+     * @return a list of junit XML results files
+     */
+    private List<File> findJunitResultFiles() {
```

Comment:
These methods appear to be in a fairly random order. Could we reorder them to that they make some logic sense when read  top to bottom?

Reply:
I tried that but gave up and went with alphabetical (non-static methods first)

Reply:
:nauseated_face: